queries.sql

```diff
@@ -8,3 +8,61 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
  SELECT * from animals WHERE neutered = TRUE;
  SELECT * from animals WHERE name != 'Gabumon';
  SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+ 
+-- begin transaction
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+-- verfying that change was made
+SELECT * FROM animals;
+
+--Roll back the change 
+ROLLBACK;
+
+-- verifying that the species columns went back to the state before the transaction.
+SELECT * FROM animals;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+BEGIN;
+
+--Deleting all records
+DELETE FROM animals;
+
+-- rolling back the transcation
+ROLLBACK;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT SP1;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SP1;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT MAX(escape_attempts), neutered FROM animals GROUP BY neutered;
+
+SELECT MIN(weight_kg) FROM animals;
+
+SELECT MAX(weight_kg) FROM animals;
```

Comment:
- [ ] While doing queries it's really useful to compare data and to get data. Kindly make these two queries in a single one to have a table with two values and can see easily the difference 🥇

Reply:
@BregornOriginal Issue Resolved